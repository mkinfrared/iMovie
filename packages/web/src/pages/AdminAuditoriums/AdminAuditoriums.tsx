import Typography from "@material-ui/core/Typography";
import AddIcon from "@material-ui/icons/Add";
import React from "react";
import { useSelector } from "react-redux";
import { useHistory, useParams } from "react-router-dom";

import AdminAuditoriumsList from "containers/AdminAuditoriumsList";
import { AdminRoutes } from "routes/Routes.type";
import { selectCinema } from "store/reducers/cinemas/selectors";
import FloatButton from "ui/FloatButton";

import css from "./AdminAuditoriums.module.scss";

type Params = {
  cinemaId: string;
};

const AdminAuditoriums = () => {
  const history = useHistory();
  const { cinemaId } = useParams<Params>();
  const id = parseInt(cinemaId, 0);
  const cinema = useSelector(selectCinema(id));

  const handleAddClick = () => {
    history.push(`${AdminRoutes.ADMIN_CINEMA}/${cinemaId}/add`);
  };

  return (
    <>
      <div className={css.AdminAuditoriums}>
        <Typography variant="h3" align="center">
          {cinema?.name} Auditoriums
        </Typography>
        <AdminAuditoriumsList cinemaId={id} />
      </div>
      <FloatButton onClick={handleAddClick} data-testid="floatButton">
        <AddIcon />
      </FloatButton>
    </>
  );
};

export default React.memo(AdminAuditoriums);
