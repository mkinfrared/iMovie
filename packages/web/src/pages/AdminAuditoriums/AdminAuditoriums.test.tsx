import { render } from "@testing-library/react";
import React, { ReactElement } from "react";
import "react-redux";

import AdminAuditoriums from "./AdminAuditoriums";

jest.mock("react-router-dom", () => ({
  useParams: jest.fn(() => "42"),
  useHistory: jest.fn()
}));

jest.mock("react-redux", () => ({
  useSelector: jest.fn(() => "42")
}));

jest.mock("containers/AdminAuditoriumsList", () => (props: any) => (
  <div data-testid="containers/AdminAuditoriumsList">
    <code>{JSON.stringify(props)}</code>
  </div>
));

describe("<AdminAuditoriums />", () => {
  let Component: ReactElement;

  beforeEach(() => {
    Component = <AdminAuditoriums />;
  });

  it("should be defined", () => {
    const { container } = render(Component);

    expect(container).toBeDefined();
  });

  it("should match a snapshot", () => {
    const { container } = render(Component);

    expect(container).toMatchSnapshot();
  });
});
