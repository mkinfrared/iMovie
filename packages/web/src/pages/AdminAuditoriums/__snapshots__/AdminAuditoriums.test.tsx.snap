// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AdminAuditoriums /> should match a snapshot 1`] = `
<div>
  <div
    class="AdminAuditoriums"
  >
    <h3
      class="MuiTypography-root MuiTypography-h3 MuiTypography-alignCenter"
    >
       Auditoriums
    </h3>
    <div
      data-testid="containers/AdminAuditoriumsList"
    >
      <code>
        {"cinemaId":null}
      </code>
    </div>
  </div>
  <div
    class="FloatButton"
  >
    <button
      class="MuiButtonBase-root MuiFab-root button MuiFab-primary"
      data-testid="floatButton"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiFab-label"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
          />
        </svg>
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;
